variables.tf

```diff
@@ -2,8 +2,150 @@
 # Input Variables
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "resource_group_id" {
+  description = "The resource group ID where the watsonx services will be provisioned. Required when creating a new instance."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.existing_watson_studio_instance_crn == null ? length(var.resource_group_id) > 0 : true
+    error_message = "You must specify a value for 'resource_group_id', required to create watson studio instance."
+  }
+
+  validation {
+    condition     = var.existing_machine_learning_instance_crn == null ? length(var.resource_group_id) > 0 : true
+    error_message = "You must specify a value for 'resource_group_id', required to create watson machine learning instance."
+  }
+}
+
+variable "prefix" {
+  description = "The name to be used on all Watson resources as a prefix."
```

Comment:
I think this can this be a better description aligning to other modules as well - `Prefix to add to all watsonx.ai resources created by this module.`

Reply:
updated